### PR TITLE
refactor: cleanup code structure

### PR DESCRIPTION
**Pull request by NoNameProvided**
This PR starts to refactor the codebase into a more maintainable structure. This is an ongoing effort and this PR won't include all the changes.

### CODE REVIEW

File: src/decorators/exclude.decorator.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { defaultMetadataStorage } from '../storage';
+import { ExcludeOptions } from '../interfaces';
+
+/**
+ * Marks property as excluded from the process of transformation. By default it excludes the property for both
```
Comment (saulotoledo) [inline src/decorators/exclude.decorator.ts:5]: I would suggest to take the chance to add to this description that this decorator may also be applied to classes.

File: src/decorators/exclude.decorator.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * constructorToPlain and plainToConstructor transformations, however you can specify on which of transformation types
+ * you want to skip this property.
+ */
+export function Exclude(options: ExcludeOptions = {}) {
```
Comment (saulotoledo) [inline src/decorators/exclude.decorator.ts:9]: We could try to add TS types for decorators. In this case `: PropertyDecorator | ClassDecorator`.

File: src/decorators/expose.decorator.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { defaultMetadataStorage } from '../storage';
+import { ExposeOptions } from '../interfaces';
+
+/**
+ * Marks property as included in the process of transformation. By default it includes the property for both
```
Comment (saulotoledo) [inline src/decorators/expose.decorator.ts]: Same here regarding the possibility to use it as classes.

File: src/decorators/expose.decorator.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+ * constructorToPlain and plainToConstructor transformations, however you can specify on which of transformation types
+ * you want to skip this property.
+ */
+export function Expose(options: ExposeOptions = {}) {
```
Comment (saulotoledo) [inline src/decorators/expose.decorator.ts]: Same here regarding the decorator type.

File: src/decorators/transform-class-to-class.decorator.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { ClassTransformer } from '../ClassTransformer';
+import { ClassTransformOptions } from '../interfaces';
+
+/**
+ * Return the class instance only with the exposed properties.
+ */
+export function TransformClassToClass(params?: ClassTransformOptions): Function {
```
Comment (saulotoledo) [inline src/decorators/transform-class-to-class.decorator.ts]: Replace function with decorator type.

File: src/decorators/transform-class-to-plain.decorator.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import { ClassTransformer } from '../ClassTransformer';
+import { ClassTransformOptions } from '../interfaces';
+
+/**
+ * Transform the object from class to plain object and return only with the exposed properties.
+ */
+export function TransformClassToPlain(params?: ClassTransformOptions): Function {
```
Comment (saulotoledo) [inline src/decorators/transform-class-to-plain.decorator.ts]: Replace function with decorator type.

File: src/decorators/transform-plain-to-class.decorator.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import { ClassTransformer } from '../ClassTransformer';
+import { ClassTransformOptions, ClassConstructor } from '../interfaces';
+
+/**
+ * Return the class instance only with the exposed properties.
+ */
+export function TransformPlainToClass<T = unknown>(
+  classType: ClassConstructor<T>,
+  params?: ClassTransformOptions
+): Function {
```
Comment (saulotoledo) [inline src/decorators/transform-plain-to-class.decorator.ts]: Replace function with decorator type.

File: src/decorators/transform.decorator.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+import { TransformOptions } from '../interfaces';
+import { TransformationType } from '../enums';
+
+/**
+ * Defines a custom logic for value transformation.
+ */
+export function Transform(
+  transformFn: (value: any, obj: any, transformationType: TransformationType) => any,
+  options: TransformOptions = {}
+) {
```
Comment (saulotoledo) [inline src/decorators/transform.decorator.ts]: Add decorator type.

File: src/decorators/type.decorator.ts [hunk 9]
```diff
@@ -0,0 +1,19 @@
+import { defaultMetadataStorage } from '../storage';
+import { TypeHelpOptions, TypeOptions } from '../interfaces';
+
+/**
+ * Specifies a type of the property.
+ * The given TypeFunction can return a constructor. A discriminator can be given in the options.
+ */
+export function Type(typeFunction?: (type?: TypeHelpOptions) => Function, options: TypeOptions = {}) {
```
Comment (saulotoledo) [inline src/decorators/type.decorator.ts]: Add decorator type.

File: src/interfaces/index.ts [hunk 10]
```diff
@@ -0,0 +1,13 @@
+export * from './decorator-options/expose-option.interface';
+export * from './decorator-options/exclude-option.interface';
+export * from './decorator-options/transform-option.interface';
+export * from './decorator-options/type-discriminator-descriptor.interface';
+export * from './decorator-options/type-option.interface';
```
Comment (saulotoledo) [inline src/interfaces/index.ts:5]: I suggest renaming these files to `.options` (with an `s`).

File: src/decorators/exclude.decorator.ts (hunk 1 above)
Comment (NoNameProvided) [reply src/decorators/exclude.decorator.ts]: Fixed.

File: src/interfaces/index.ts (hunk 10 above)
Comment (NoNameProvided) [reply src/interfaces/index.ts]: Done.

**Comment by saulotoledo** [comment]
Last review done. LGTM.

**Pull request merged**